src/content/peerconnection/audio/js/main.js

```diff
@@ -359,3 +367,33 @@ window.setInterval(() => {
     lastResult = res;
   });
 }, 1000);
+
+if (window.RTCRtpReceiver && ('getSynchronizationSources' in window.RTCRtpReceiver.prototype)) {
+  let lastTime;
+  const getAudioLevel = (timestamp) => {
+    window.requestAnimationFrame(getAudioLevel);
+    if (!pc2) {
+      return;
+    }
+    const receiver = pc2.getReceivers().find(r => r.track.kind === 'audio');
+    if (!receiver) {
+      return;
+    }
+    const sources = receiver.getSynchronizationSources();
+    sources.forEach(source => {
+      audioLevels.push(source.audioLevel);
+    });
+    if (!lastTime) {
+      lastTime = timestamp;
+    } else if (timestamp - lastTime > 500 && audioLevels.length > 0) {
+      // Update graph every 500ms.
```

Comment:
We're collecting data points for every animation frame (e.g. 30 fps?) but we're only updating the graph every 500 ms. It seems a bit strange to me to call getSSRC this often if we end up with a very rough 0.5 fps audio meter. Should we update it a bit more often or call getSSRCs no as often? No strong opinion here though

Reply:
the recommendation I heard was to use RAF together with this API.